From b34b8acf335ae10911c9b4ae226b63f164597e03 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 14 Mar 2019 03:55:56 -0700
Subject: [PATCH 0570/1851] v4l: xilinx: vpss-scaler: Make max width and height
 mandatory properties

commit 5b2f93347a919fc1c98ecf1d14cec08bc9ceb2e6 from
https://github.com/Xilinx/linux-xlnx.git

This patch makes the xlnx,max-width and xlnx,max-height as mandatory dt
properties. This breaks backward compatibility and a warning message is
hence added.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vpss-scaler.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
index 1d6f0b094ff8..21612a33222b 100644
--- a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
@@ -29,8 +29,6 @@
 #define XSCALER_MAX_WIDTH		(8192)
 #define XSCALER_MIN_HEIGHT		(64)
 #define XSCALER_MAX_HEIGHT		(4320)
-#define XSCALER_DEF_MAX_WIDTH		(3840)
-#define XSCALER_DEF_MAX_HEIGHT		(2160)
 #define XSCALER_MAX_PHASES		(64)
 
 /* Modify to defaults incase it is not configured from application */
@@ -1596,7 +1594,8 @@ static int xscaler_parse_of(struct xscaler_device *xscaler)
 	ret = of_property_read_u32(node, "xlnx,max-height",
 				   &xscaler->max_lines);
 	if (ret < 0) {
-		xscaler->max_lines = XSCALER_DEF_MAX_HEIGHT;
+		dev_err(dev, "xlnx,max-height is missing!");
+		return -EINVAL;
 	} else if (xscaler->max_lines > XSCALER_MAX_HEIGHT ||
 		   xscaler->max_lines < XSCALER_MIN_HEIGHT) {
 		dev_err(dev, "Invalid height in dt");
@@ -1606,7 +1605,8 @@ static int xscaler_parse_of(struct xscaler_device *xscaler)
 	ret = of_property_read_u32(node, "xlnx,max-width",
 				   &xscaler->max_pixels);
 	if (ret < 0) {
-		xscaler->max_pixels = XSCALER_DEF_MAX_WIDTH;
+		dev_err(dev, "xlnx,max-width is missing!");
+		return -EINVAL;
 	} else if (xscaler->max_pixels > XSCALER_MAX_WIDTH ||
 		   xscaler->max_pixels < XSCALER_MIN_WIDTH) {
 		dev_err(dev, "Invalid width in dt");
-- 
2.31.1

